### DISCUSSION TITLE
Fix the problem of "ROC-RK3588S-PC" fan full-speed run

### DISCUSSION THREAD

**Discussion by travel-star** [Joshua-Riek/ubuntu-rockchip#558] [Show and tell]
1.add firefly source list into  /etc/apt/sources.list.d
deb http://wiki.t-firefly.com/firefly-ubuntu-repo jammy main
deb http://wiki.t-firefly.com/firefly-rk3588-repo focal main
cp trusted.gpg into /etc/apt 
[trusted.zip](https://github.com/Joshua-Riek/ubuntu-rockchip/files/13868125/trusted.zip)

2.install firefly's firefly-fan
apt-get update
apt-get upgrade
apt-get install firefly-fan

3.change /usr/bin/firefly-fan-init(the init program can‘t correct identification the board model)
line 5
change"RK3588_BOARD_VERSION=`cat /sys/firmware/devicetree/base/model | awk -F ' ' {'print $2'}`"to"RK3588_BOARD_VERSION=`cat /sys/firmware/devicetree/base/model | awk -F ' ' {'print $1'}`"
![image](https://github.com/Joshua-Riek/ubuntu-rockchip/assets/32922240/1a3e6a06-848b-41ee-850f-79c34f31ecc7)

4.
restart firefly-fan
systemctl restart firefly-fan.service

Wooo! this word is so silent.

**Comment by Joshua-Riek** [comment]
I wonder why there are not any default cooling levels in the device tree to prevent the fan from going full-blast. I would add something like the below code to the DTS. But I don't have the board to test this out.
```diff
diff --git a/arch/arm64/boot/dts/rockchip/rk3588s-roc-rk3588s-pc-v12.dts b/arch/arm64/boot/dts/rockchip/rk3588s-roc-rk3588s-pc-v12.dts
index 37e9c5203c91..e97a45d3d870 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588s-roc-rk3588s-pc-v12.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3588s-roc-rk3588s-pc-v12.dts
@@ -81,9 +81,18 @@ es8388_sound: es8388-sound {
 
        fan: pwm-fan {
                compatible = "pwm-fan";
+               cooling-levels = <100 160 190 200 215 235 255>;
                #cooling-cells = <2>;
                fan-supply = <&vcc12v_dcin>;
                pwms = <&pwm11 0 50000 1>;
+               rockchip,temp-trips = <
+                       55000   1
+                       60000   2
+                       65000   3
+                       70000   4
+                       75000   5
+                       80000   6
+               >;
        };
 
        firefly_leds: leds {
```
